django/core/servers/basehttp.py

```diff
@@ -227,13 +227,30 @@ def handle_one_request(self):
         handler.run(self.server.get_app())
 
 
-def run(addr, port, wsgi_handler, ipv6=False, threading=False, server_cls=WSGIServer):
+def run(
+    addr,
+    port,
+    wsgi_handler,
+    ipv6=False,
+    threading=False,
+    server_cls=WSGIServer,
+    on_bind=None,
+):
     server_address = (addr, port)
-    if threading:
-        httpd_cls = type("WSGIServer", (socketserver.ThreadingMixIn, server_cls), {})
-    else:
-        httpd_cls = server_cls
+    cls_dict = {}
+
+    if on_bind is not None:
+
+        def server_bind(self):
+            server_cls.server_bind(self)
+            on_bind(self.server_port)
+
+    cls_dict["server_bind"] = server_bind
```

Comment:
It looks to me as if `server_bind` would be referenced before  assignment when `on_bind is None`. 

Can you please add a test case for that scenario, or else we'd need to make the parameter required if it were always passed. 🤔

Reply:
Apparently, on_bind is always passed, so I made it a required parameter.

Reply:
Hi, @carltongibson sorry for the confusion for some reason my changes where I made it a required weren't pushed. 

However, I have made `on_bind` a required argument now.